feat: implement trading button visibility logic and related hooks

originalix opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Unified “Enable Trading” button logic across Perp trading views; shown during account loading or when trading isn’t available.
  * On mobile, TP/SL controls are hidden until trading is enabled.
  * Mobile Order Book depth now adapts: 4 levels when “Enable Trading” is shown, 10 with TP/SL enabled, otherwise 8.

* **Style**
  * Updated TP/SL trigger label text while keeping existing tooltip content.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

originalix committed:
feat: implement trading button visibility logic and related hooks

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/e55b1e14-e656-4810-ac4b-8279acf30bea?refs=596423f9698d860ec6024b35bfcd335fcdc553bd&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/e55b1e14-e656-4810-ac4b-8279acf30bea/license?refs=596423f9698d860ec6024b35bfcd335fcdc553bd&source=prChecksComment)

originalix committed:
feat: adjust mobile max levels per side based on trading button visibility

# Conflicts:
#	packages/kit/src/views/Perp/components/TradingPanel/panels/PerpTradingForm.tsx

originalix committed:
feat: conditionally render TP/SL inputs based on trading button visibility for mobile

sidmorizon commented on packages/kit/src/views/Perp/hooks/usePerpEnableTradingStatus.ts:
```diff
@@ -0,0 +1,40 @@
+} from '@onekeyhq/kit-bg/src/states/jotai/atoms';
+
+export function usePerpShouldShowEnableTradingButton() {
+  const [shouldShowEnableTradingButton] =
```
不要套壳封装，业务上直接用 usePerpsShouldShowEnableTradingButtonAtom

originalix replied:
ok，业务上直接用 atom 吧

sidmorizon commented on packages/kit/src/views/Perp/hooks/usePerpEnableTradingStatus.ts:
```diff
@@ -0,0 +1,40 @@
+  return shouldShowEnableTradingButton;
+}
+
+export function usePerpEnableTradingStatus() {
```
不要套壳封装，业务上需要哪个用哪个

originalix replied:
这个本来是从业务上抽出来准备复用 shouldShowEnableTradingButton 的逻辑的

但观察到依赖的 atom 过多，可能会让 orderbook 频繁 re-render，才新增了一个 atom 只返回 boolean 类型

其实这个 hooks 也不需要了，我把业务代码 revert 回去

sidmorizon commented on packages/kit/src/views/Perp/hooks/usePerpEnableTradingStatus.ts:
```diff
@@ -0,0 +1,40 @@
+  const [perpsAccountStatus] = usePerpsActiveAccountStatusAtom();
+  const [perpsAccountLoading] = usePerpsAccountLoadingInfoAtom();
+
+  const isAccountLoading = useMemo(() => {
```
可以封装 computed

originalix committed:
fix: remove usePerpEnableTradingStatus

originalix committed:
fix: remove unused platformEnv import from PerpTradingForm
